packages/endomoat/src/cli.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Main CLI entry point
+ *
+ * @remarks
+ * As tempting as it may be to try to move stuff out of here, it _will_ break
+ * type inference, and you'll need to sort that out yourself.
+ *
+ * Regarding middleware: Any option which _does not_ either a) need other
+ * options for postprocessing, or b) need to be asynchronous--should use
+ * {@link yargs.coerce} instead.
+ * @packageDocumentation
+ */
+
+import path from 'node:path'
+import yargs from 'yargs'
+import { hideBin } from 'yargs/helpers'
+import * as constants from './constants.js'
+import { absolutify, resolveFrom } from './util.js'
+
+yargs(hideBin(process.argv))
+  /**
+   * Default command (no command)
+   */
+  .command(
+    '$0 [entryPath]',
+    'Start the application',
+    (yargs) =>
+      yargs
+        .options({
+          policy: {
+            alias: ['p', 'policyPath'],
+            describe:
+              'Filepath to the policy file, relative to the application directory',
+            type: 'string',
+            normalize: true,
+            default: constants.DEFAULT_POLICY_PATH,
+          },
+          'policy-override': {
+            alias: ['o', 'override'],
+            describe:
+              'Filepath to the policy override file, relative to the application directory',
+            type: 'string',
+            normalize: true,
+            default: constants.DEFAULT_POLICY_OVERRIDE_PATH,
+          },
+          'endo-policy': {
+            alias: ['e', 'endo'],
+            describe:
+              'Path to an Endo policy file, relative to application directory',
+            normalize: true,
+            conflicts: ['policy', 'policyOverride'],
+          },
+          cwd: {
+            describe: 'The application directory',
+            type: 'string',
+            normalize: true,
+            default: process.cwd(),
+            defaultDescription: '(current directory)',
+            // force absolute path
+            coerce: path.resolve,
+          },
+        })
+        .positional(
+          'entryPath',
+          /**
+           * @remarks
+           * The `default` property value _must_ be a string, or TS will think
+           * it can be `undefined`. If it's an empty string, the
+           * `resolveEntryPath` middleware will attempt to resolve it
+           * automatically.
+           */
+          {
+            describe: 'Path to the application entry point',
+            type: 'string',
+            normalize: false,
+            defaultDescription: '(derived from package.json)',
+            default: '',
+          }
+        ),
+    /**
+     * Default command handler.
+     *
+     * @remarks
+     * (That's "handler for the default command"--_not_ "default handler for a
+     * command").
+     */
+    async (argv) => {
+      const { run } = await import('./index.js')
```

Comment:
Why do we need dynamic imports here? It's fine for them to be in a node specific cli, but we tended to avoid them in code because none of our tools support them.

Reply:
premature optimization, I suppose.  there's no need to load `index.js` if the arg parsing fails because a user provided a bad option.